src/large_scale_condensation.jl

```diff
@@ -1,46 +1,99 @@
-function get_large_scale_condensation_tendencies!(
-    Diag::DiagnosticVariables{NF},
-    M   ::ModelSetup,
-) where {NF<:AbstractFloat}
-    @unpack gravity, RH_thresh_max, RH_thresh_range, RH_thresh_boundary, humid_relax_time = M.constants
-    @unpack cp, alhc, k_lsc = M.parameters
-    @unpack nlon, nlat, nlev, σ_levels_full, σ_levels_thick = M.geospectral.geometry
-    @unpack temp_grid, humid_grid, pres_grid = Diag.grid_variables
-    @unpack temp_tend_lsc, humid_tend_lsc, sat_spec_humidity, sat_vap_pressure, cloud_top, precip_large_scale = Diag.parametrization_variables
-
-    get_saturation_specific_humidity!(sat_spec_humidity, sat_vap_pressure, temp_grid, pres_grid, M)
+function large_scale_condensation!( column::ColumnVariables{NF},
+                                    model::PrimitiveEquationModel,
+                                    ) where {NF<:AbstractFloat}
+
+    get_saturation_vapour_pressure!(column, model)
+    get_saturation_specific_humidity!(column, model)
+    
+    @unpack gravity, RH_thresh_max, RH_thresh_range, RH_thresh_boundary, humid_relax_time = model.constants
+    @unpack cp, alhc, k_lsc = model.parameters
+    @unpack σ_levels_full, σ_levels_thick = model.geometry
+
+    @unpack humid, pres = column            # prognostic variables: specific humidity, surface pressure
+    @unpack temp_tend, humid_tend = column  # tendencies to write into
 
     # 1. Tendencies of humidity and temperature due to large-scale condensation
-    for k = k_lsc:nlev  # Used to be 2:nlev in original Speedy
-        σₖ = σ_levels_full[k]
-        RH_threshold = RH_thresh_max + RH_thresh_range * (σₖ^2 - 1)  # Relative humidity threshold for condensation (Formula 24)
+    # skip the planetary boundary layer, k_lsc is lowest level for precip 
+    for k in eachlayer(column)[k_lsc:end]
+        
+        # Relative humidity threshold for condensation (Formula 24)
+        σₖ² = σ_levels_full[k]^2
+        RH_threshold = RH_thresh_max + RH_thresh_range * (σₖ² - 1)
         if k == nlev
             RH_threshold = max(RH_threshold, RH_thresh_boundary)
         end
```

Comment:
These seem to be constants, for now fine, but we can think about precomputing `RH_threshold` somewhere